File: 1047-remove-all-adjacent-duplicates-in-string/1047-remove-all-adjacent-duplicates-in-string.cpp
class Solution {
public:
        string removeDuplicates(string s) {

        string ans = s;

        int n = ans.size();
        for(int i=0;i<n-1;){
            if(ans[i]==ans[i+1]){
                //cout<<i<<" "<<i+1<<" "<<n<<endl;
                ans.erase(i,1);
                ans.erase(i,1);
                n-=2;
                if(i!=0) i--;
                //cout<<i<<" "<<i+1<<" "<<n<<endl;
            }
            else{
                i++;
            }
        }

        return ans;
    }
};

File: 1047-remove-all-adjacent-duplicates-in-string/1047-remove-all-adjacent-duplicates-in-string.cs
public class Solution {
    public string RemoveDuplicates(string s) {
        Stack<char> stack = new Stack<char>();
        //stack.Push(s[0]);
        
        foreach(var c in s)
        {
            if(stack.Any() && stack.Peek() == c)
            {
                while(stack.Any() && stack.Peek() == c)
                {
                    stack.Pop();
                }
            }
            else
            {
                stack.Push(c);
            }
        }
        string ans = "";
        while(stack.Any())
        {
            ans = stack.Pop() + ans;
        }

        return ans;
    }
}